### ISSUE TITLE
"Password is not set up" after successful login

### ISSUE THREAD

**Issue by jean-christophe-manciot** [keycloak/keycloak#22126]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

account/ui

### Describe the bug

After a successful initial login, accessing the admin account shows a strange "Password is not set up" error.
Then, after logging out, it is not possible to log in again.

### Version

22.0.1

### Expected behavior

No "Password is not set up" error

### Actual behavior

The admin user has been created remotely during the initial start with:
```
export KEYCLOAK_ADMIN=<username>
export KEYCLOAK_ADMIN_PASSWORD=<password>
```

After a successful initial admin login, the "Signing in" tab of the admin account shows:
![Screenshot from 2023-07-31 16-37-43](https://github.com/keycloak/keycloak/assets/13176858/d64d1476-98b0-44a5-8584-6c0512a5beb1)

Furthermore, it is not possible to remedy this situation as trying to setup a new password fails because this old one is not ... recognized.

Then, logging out and logging back in leads to:
![Screenshot from 2023-07-31 17-37-05](https://github.com/keycloak/keycloak/assets/13176858/1fed23db-382e-40da-afa7-156acf2f6bba)

### How to Reproduce?

On keycloak server 22.0.1:
Building with:
```
bin/kc.sh build --cache-config-file="cache-ispn-tcpping.xml"
```
and starting with:
```
KEYCLOAK_ADMIN=admin
KEYCLOAK_ADMIN_PASSWORD='my-password' 
KC_EXTERNAL_ADDR=192.168.122.21 
KC_BIND_ADDR=192.168.122.21 
KC_BIND_PORT=7800 
KC_INITIAL_HOSTS='"192.168.122.21[7800],192.168.122.22[7800]"'

bin/kc.sh start --optimized &
```

**Comment by ssilvert** [comment]
Do you get the same problem if you set up the initial admin password using the web page?  (i.e. localhost:8080/)

**Comment by ssilvert** [comment]
Also, can you give it a try with the nightly build?  This will run `keycloak.v3` account console by default instead of `keycloak.v2`.

[latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

**Comment by jean-christophe-manciot** [comment]
@ssilvert 

Each installation detailed below is performed after:
- purging the keycloak server 
- purging the whole keycloak database on the external MySQL server.

1. Installing **22.0.1** and setting the admin's password remotely **without any special characters**
The password is considered as created:
![Screenshot from 2023-08-01 15-08-08](https://github.com/keycloak/keycloak/assets/13176858/40cb43e5-315a-4e3a-a431-4058b85f4092)

2. Installing **nightly** and setting the admin's password remotely **with some special characters**
The password is considered as created:
![Screenshot from 2023-08-01 15-54-53](https://github.com/keycloak/keycloak/assets/13176858/ff5a80a2-ed9a-49bb-ac30-764a5d8b2ed1)

**Comment by ssilvert** [comment]
Can you try switching the theme on nightly to `keycloak.v2`?  I'd like to know conclusively if it's the account console version or the server version.

**Comment by jean-christophe-manciot** [comment]
> Can you try switching the theme on nightly to `keycloak.v2`?

How?

**Comment by ssilvert** [comment]
@jean-christophe-manciot 

Go to `Realm settings --> Themes`
Set `Account Theme` to `keycloak.v2`

**Comment by jean-christophe-manciot** [comment]
> Go to `Realm settings --> Themes` Set `Account Theme` to `keycloak.v2`
![Screenshot from 2023-08-02 14-53-57](https://github.com/keycloak/keycloak/assets/13176858/104271be-70fa-42d7-b4a1-afaad07aafdf)

The admin's password is still considered as created:
![Screenshot from 2023-08-02 14-55-10](https://github.com/keycloak/keycloak/assets/13176858/1aa889aa-e5e9-426f-874b-9818b9178739)

**Comment by ssilvert** [comment]
OK.  That means whatever the problem was, it appears to be fixed on the server side in the nightly build.  And it's not likely to be a UI problem.

Do you concur that this is now fixed?  If so, I'd like to close this for now.  We can always reopen if you see the problem again.

**Comment by jean-christophe-manciot** [comment]
I agree that the fix to some special character(s) flaw did not come from  the upgrade keycloak --> keycloak.v3.
